Maker-Agents-Suit-Module/OSMAgent/src/rely.function.spec.ts

```diff
@@ -1,142 +1,95 @@
+    const mockFetcher: any = { get: jest.fn() };
     handleTransaction = provideRelyFunctionHandler(mockFetcher);
 
     when(mockFetcher.get).calledWith(1).mockReturnValue(CONTRACTS[1]);
     when(mockFetcher.get).calledWith(2).mockReturnValue(CONTRACTS[2]);
     when(mockFetcher.get).calledWith(3).mockReturnValue(CONTRACTS[3]);
   });
 
-  it("should return a finding for one of the OSM contract", async () => {
+  it("should return a finding for rely call on an OSM contract", async () => {
     const _from = createAddress("0x2");
-    const _to = CONTRACTS[1][0]; 
-    const _input: string = ABI.encodeFunctionCall(
-      {
-        name: "rely",
-        type: "function",
-        inputs: [
-          {
-            type: "address",
-            name: "usr",
-          },
-        ],
-      },
-      [ADDRESS]
-    );
+    const _to = CONTRACTS[1][0];
+    const _input: string = relyIface.encodeFunctionData("rely", [ADDRESSES[0]]);
 
-    const txEvent: TransactionEvent = new TestTransactionEvent()
-      .setTimestamp(1)
-      .addTraces({
-        to: _to,
-        from: _from,
-        input: _input,
-      });
+    const txEvent: TransactionEvent = new TestTransactionEvent().setTimestamp(1).setTo(_to).addTraces({
+      to: _to,
+      from: _from,
+      input: _input,
+    });
 
     const findings: Finding[] = await handleTransaction(txEvent);
-    expect(findings).toStrictEqual([createFinding(_to, ADDRESS)]);
+    expect(findings).toStrictEqual([createFinding(_to, ADDRESSES[0])]);
   });
 
   it("should return multiple findings", async () => {
     const _from = createAddress("0x2");
-    const _input: string = ABI.encodeFunctionCall(
-      {
-        name: "rely",
-        type: "function",
-        inputs: [
-          {
-            type: "address",
-            name: "usr",
-          },
-        ],
-      },
-      [ADDRESS]
-    );
+    const _input: string = relyIface.encodeFunctionData("rely", [ADDRESSES[0]]);
+    const _input2: string = relyIface.encodeFunctionData("rely", [ADDRESSES[1]]);
 
-    const txEvent: TransactionEvent = new TestTransactionEvent()
-      .setTimestamp(2)
-      .addTraces({
+    const txEvent: TransactionEvent = new TestTransactionEvent().setTimestamp(2).setTo(CONTRACTS[2][2]).addTraces(
+      {
         to: CONTRACTS[2][2],
         from: _from,
         input: _input,
-      },{
-        to: CONTRACTS[2][1],
+      },
+      {
+        to: CONTRACTS[2][2],
         from: _from,
-        input: _input,
-      });
+        input: _input2,
+      }
+    );
 
     const findings: Finding[] = await handleTransaction(txEvent);
     expect(findings).toStrictEqual([
-      createFinding(CONTRACTS[2][1], ADDRESS),
-      createFinding(CONTRACTS[2][2], ADDRESS),
+      createFinding(CONTRACTS[2][2], ADDRESSES[0]),
+      createFinding(CONTRACTS[2][2], ADDRESSES[1]),
     ]);
   });
 
   it("should return empty finding when OSM contract address does found", async () => {
```

Comment:
```suggestion
  it("should return an empty finding when OSM contract address is not found", async () => {
```
"is not found" is another guess at what was meant.